source/core_http_client.c

```diff
@@ -553,6 +569,28 @@ static uint32_t getZeroTimestampMs( void )
 
 /*-----------------------------------------------------------*/
 
+static int stringCaseCmp( const char * str1,
```

Comment:
I think it is a MISRA violation if you use `int` instead of specifying how many bits it should be. Also, probably best to just use a single byte.
```suggestion
static int8_t stringCaseCmp( const char * str1,
```